virttest/qemu_vm.py

```diff
@@ -2855,6 +2855,19 @@ def create(self, name=None, params=None, root_dir=None,
                 # Add this monitor to the list
                 self.monitors.append(monitor)
 
+            # Create consoles in advance, since they should be
+            # ready before sending 'cont'
+            self.create_serial_console()
+
+            for key, value in self.logs.items():
+                outfile = "%s-%s.log" % (key, name)
+                self.logsessions[key] = aexpect.Tail(
+                    "nc -U %s" % value,
+                    auto_close=False,
+                    output_func=utils_misc.log_line,
+                    output_params=(outfile,))
+                self.logsessions[key].set_log_file(outfile)
+
```

Comment:
there is already same code logic available just before vm `cont`, why this again needed?
https://github.com/avocado-framework/avocado-vt/blob/b1aa0b794a8a508d9a6bc905a46e1b7c0c9ddd27/virttest/qemu_vm.py#L2922
```
            self.vhost_threads = self.get_vhost_threads(vhost_thread_pattern)

            self.create_serial_console()

            for key, value in list(self.logs.items()):
                outfile = "%s-%s.log" % (key, name)
                self.logsessions[key] = aexpect.Tail(
                    "nc -U %s" % value,
                    auto_close=False,
                    output_func=utils_misc.log_line,
                    output_params=(outfile,))
                self.logsessions[key].set_log_file(outfile)

            if params.get("paused_after_start_vm") != "yes":
                # start guest
                if self.monitor.verify_status("paused"):
                    if not migration_mode:
                        self.monitor.cmd("cont")
```

Reply:
Yes, this logic is already available, the consoles are created in subprocesses and just before sending 'cont'. But we don't know if the console connections are done or not before sending 'cont'. In in our scenario, when there is a problem in OVMF and it can't boot into guest os, the consoles seems not grub the whole ovmf log and missed the head part, so I want to advance this logic code far before sending 'cont' to ensure the consoles connection has built successfully before sending 'cont'.